packages/ui-patterns/src/FilterBar/FilterBarContext.tsx

```diff
@@ -0,0 +1,309 @@
+'use client'
+
+import React, { createContext, useCallback, useContext, useRef } from 'react'
+import { ActiveInput, useFilterBarState, useOptionsCache } from './hooks'
+import { MenuItem } from './menuItems'
+import { FilterBarAction, FilterGroup, FilterProperty } from './types'
+import { useCommandHandling } from './useCommandHandling'
+import { useKeyboardNavigation } from './useKeyboardNavigation'
+import {
+  findConditionByPath,
+  isAsyncOptionsFunction,
+  removeFromGroup,
+  updateNestedLogicalOperator,
+  updateNestedOperator,
+  updateNestedValue,
+} from './utils'
+
+export type FilterBarContextValue = {
+  // Core state
+  filters: FilterGroup
+  filterProperties: FilterProperty[]
+  activeInput: ActiveInput
+  freeformText: string
+  isLoading: boolean
+  error: string | null
+
+  // Handlers
+  onFilterChange: (filters: FilterGroup) => void
+  onFreeformTextChange: (text: string) => void
+  setActiveInput: (input: ActiveInput) => void
+  handleInputChange: (path: number[], value: string) => void
+  handleOperatorChange: (path: number[], value: string) => void
+  handleRemoveCondition: (path: number[]) => void
+  handleKeyDown: (e: React.KeyboardEvent<HTMLInputElement>) => void
+  handleSelectMenuItem: (item: MenuItem) => void
+  handleInputFocus: (path: number[]) => void
+  handleOperatorFocus: (path: number[]) => void
+  handleGroupFreeformFocus: (path: number[]) => void
+  handleInputBlur: () => void
+  handleGroupFreeformChange: (path: number[], value: string) => void
+  handleLabelClick: (path: number[]) => void
+  handleLogicalOperatorChange: (path: number[]) => void
+
+  // Options cache
+  propertyOptionsCache: Record<string, { options: any[]; searchValue: string }>
```

Comment:
Trying to avoid `any` here, I know this is coming from FilterCondition but it might be a good place to clean this up. Any just causes so many headaches for developers trying to debug this in the future and since we are modifying these files we should try to clean this up for others